caddyhttp/markdown/template.go

```diff
@@ -51,25 +55,67 @@ func execTemplate(c *Config, mdata metadata.Metadata, meta map[string]string, fi
 		Meta:    meta,
 		Files:   files,
 	}
-
 	templateName := mdata.Template
-	// reload template on every request for now
-	// TODO: cache templates by a general plugin
-	if templateFile, ok := c.TemplateFiles[templateName]; ok {
-		err := SetTemplate(c.Template, templateName, templateFile)
+
+	updateTemplate := func() error {
+		templateUpdateMu.Lock()
+		defer templateUpdateMu.Unlock()
+
+		templateFile, ok := c.TemplateFiles[templateName]
+		if !ok {
+			return nil
+		}
+
+		currentFileInfo, err := os.Lstat(templateFile.path)
+		if err != nil {
+			return err
+		}
+
+		if !fileChanged(currentFileInfo, templateFile.fi) {
+			return nil
+		}
+
+		// update template due to file changes
+		err = SetTemplate(c.Template, templateName, templateFile.path)
 		if err != nil {
-			return nil, err
+			return err
 		}
+
+		templateFile.fi = currentFileInfo
+		return nil
+	}
+
+	// reload template on every request for now
+	// TODO: cache templates by a general plugin
```

Comment:
Is this TODO still needed? This seems "good enough" as a caching mechanism, I think.

Reply:
Hmm, I think we could remove this now.